assets/js/googlesitekit/datastore/user/authentication.js

```diff
@@ -81,6 +81,24 @@ const baseSelectors = {
 		return authentication;
 	},
 
+	/**
+	 * Checks to see if the current user has a particular scope.
+	 *
+	 * Returns `undefined` if the scope info is not available/loaded.
+	 *
+	 * @private
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {string} scope The scope constant to check for.
+	 * @return {(boolean|undefined)} `true` if scope is present; `false` if not.
+	 */
+	hasScope: createRegistrySelector( ( select ) => ( state, scope ) => {
+		const grantedScopes = select( STORE_NAME ).getGrantedScopes( state );
+
+		return ( grantedScopes !== undefined ) ? grantedScopes.includes( scope ) : undefined;
```

Comment:
Nit-picking, but I think using a regular `if` clause for checking and returning undefined is easier to follow than this ternary one.